controller/src/main/java/io/pravega/controller/server/rpc/auth/AuthHelper.java

```diff
@@ -10,25 +10,32 @@
 package io.pravega.controller.server.rpc.auth;
 
 import com.google.common.annotations.VisibleForTesting;
-import io.jsonwebtoken.Jwts;
-import io.jsonwebtoken.SignatureAlgorithm;
 import io.pravega.auth.AuthHandler;
 import io.pravega.auth.AuthorizationException;
+import io.pravega.shared.security.token.JsonWebToken;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
+
 import lombok.AllArgsConstructor;
 
 /**
  * Helper class containing APIs related to delegation token and authorization/authentication.
  */
 @AllArgsConstructor
 public class AuthHelper {
+
     private final boolean isAuthEnabled;
     private final String tokenSigningKey;
+    private final Optional<Integer> accessTokenTtlInSeconds;
```

Comment:
I actually started out with using the `Integer` object here, as well as other places where the TTL is being defined/used, but realized that code wasn't revealing its intent well. Also, the null checks were confusing. 

I then modified the code to use Optional to clearly signal that a missing value is permitted and not depending on a value being null. It also allows for disambiguating between these scenarios (as shown in `JsonWebToken`): 
* An invalid argument null for the `Optional<Integer>` TTL parameter
* A valid input containing empty instance for the same (say, `Optional.empty()`)
* A valid input containing a non-empty instance for the same (say, `Optional.of(200)`)

The extra constructor was a remnant from my intermediate work using `Integer`, and was merely to avoid having to modify some of the tests when I went from using `Integer` to `Optional<Integer>`; I've fixed the tests now instead and removed the redundant constructor.

Reply:
`Optional<>` as a method argument is an anti-pattern.

Please refer to this doc as to what the purpose of `Optional<>` is: https://community.oracle.com/docs/DOC-991686

A simple google search for `best practices with java optional` will show a good number of other documents and blog posts explaining that `Optional<>` as a method arg is a bad idea.

Consider the very good example listed as #15. here: https://www.baeldung.com/java-optional

Reply:
Thanks for pointing that out. I didn't realize that there are strong opinions in the community against using Optional as method arguments. There are a number of arguments laid out against using it as method arguments in [this](https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments) discussion. There are also arguments made for using it by some ([example](https://dzone.com/articles/optional-method-parameters)). As for this case, I don't have a strong opinion for or against it - it was merely about a slightly better-looking code. I'll use the `Integer` instead.

Reply:
I've pushed a commit with the change, so resolving the conversation now.